ThirdPartyAdapters/vungle/vungle/src/main/java/com/vungle/mediation/VungleBannerAdapter.java

```diff
@@ -0,0 +1,265 @@
+package com.vungle.mediation;
+
+import android.content.Context;
+import android.util.Log;
+import android.view.View;
+import android.widget.RelativeLayout;
+
+import androidx.annotation.NonNull;
+import androidx.annotation.Nullable;
+
+import com.google.ads.mediation.vungle.VungleInitializer;
+import com.google.android.gms.ads.AdRequest;
+import com.vungle.warren.AdConfig;
+import com.vungle.warren.Banners;
+import com.vungle.warren.LoadAdCallback;
+import com.vungle.warren.PlayAdCallback;
+import com.vungle.warren.Vungle;
+import com.vungle.warren.VungleBanner;
+import com.vungle.warren.VungleNativeAd;
+import com.vungle.warren.error.VungleException;
+
+import java.util.concurrent.atomic.AtomicBoolean;
+
+class VungleBannerAdapter {
+    private static final String TAG = VungleInterstitialAdapter.TAG;
+
+    @NonNull
+    private String mPlacementId;
+    @Nullable
+    private String mUniquePubRequestId;
+    @Nullable
+    private VungleListener mVungleListener;
+    @NonNull
+    private RelativeLayout mAdLayout;
+    @NonNull
+    private AdConfig mAdConfig;
+
+    private AtomicBoolean mIsLoading = new AtomicBoolean(true);
+    private boolean mPendingRequestBanner = false;
+    private boolean mVisibility = true;
+    @Nullable
+    private VungleBanner mVungleBannerAd;
+    @Nullable
+    private VungleNativeAd mVungleNativeAd;
+
+    @NonNull
+    private VungleManager mVungleManager;
+
+    VungleBannerAdapter(@NonNull String placementId, @Nullable String uniquePubRequestId) {
+        mVungleManager = VungleManager.getInstance();
+        this.mPlacementId = placementId;
+        this.mUniquePubRequestId = uniquePubRequestId;
+    }
+
+    @NonNull
+    String getPlacementId() {
+        return mPlacementId;
+    }
+
+    @Nullable
+    String getUniquePubRequestId() {
+        return mUniquePubRequestId;
+    }
+
+    boolean isLoading() {
+        return mIsLoading.get();
+    }
+
+    void setVungleListener(@Nullable VungleListener vungleListener) {
+        this.mVungleListener = vungleListener;
+    }
+
+    void setAdLayout(@NonNull RelativeLayout adLayout) {
+        this.mAdLayout = adLayout;
+    }
+
+    void setAdConfig(@NonNull AdConfig adConfig) {
+        this.mAdConfig = adConfig;
+    }
+
+    void requestBannerAd(@NonNull Context context, @NonNull String appId) {
+        mIsLoading.set(true);
+        mPendingRequestBanner = true;
+        VungleInitializer.getInstance().initialize(appId, context.getApplicationContext(),
```

Comment:
Not related to these changes, but observing Vungle.isInitialized() returning false on banner refresh